### ISSUE TITLE
let user set icon view order (keyword or upload date) durably

### ISSUE THREAD

**Issue by kaberett** [dreamwidth/dw-free#1662]
(Was Bugzilla 4949)

**Suggestion:** http://dw-suggestions.dreamwidth.org/1291181.html

**Summary:** There are two options for sorting icons on the "Manage Icons" and "Icons"
pages: by keyword and by upload date/time. Both pages default to upload date.
Let the user set a preferred order under Settings.

**Description:** On the "Manage Icons" page, http://www.dreamwidth.org/editicons, and the
"Icons" page, http://USERID.dreamwidth.org/icons, there are two options for
sorting the icons: keyword and upload date. Both default to Upload date. I have
little use for upload order, and it's annoying to have to click "Keyword order"
every single time I reopen the page, which has been about a dozen times today
(while uploading icons from LJ, which provides only upload order; kudos to you
for offering a choice!).

Let the user choose Keyword order or Upload order as a durable setting. In
fact, it could be useful to have two separate durable order settings, one for
each of those pages. 

**Suggested by:** thnidu

**Implementation note (by @rahaeli):** don't use a new setting; instead make it remember the
last-used and use that. should probably also apply to icon browser.

**Comment by kaberett** [comment]
... huh. We actually have a setting for this, sort of, under Customize Journal Style->Presentation. The thing is, per https://www.dreamwidth.org/support/see_request?id=32912 it doesn't see to... actually *do* anything.

@kareila, @rahaeli, et al -- is this "file a new bug" territory?

**Comment by kareila** [comment]
If it is a journal style property, it only applies to the logged-in user's own journal, and probably only applies when viewing the icons page in journal style, as opposed to site style, which is still the default most everywhere. 

Whoever takes this up will have to figure out how to make the journal style setting contend with the user's preferred sort order in that use case.  The wording of the style option is confusing enough that I wonder if it shouldn't be changed to something like "default sort order for this journal's custom icons page".

**Comment by kaberett** [comment]
The journal style property isn't even being effectively applied to a logged-in user's own icons page displayed in their journal style, as far as I can tell -- I've just tested on my own account -- hence wondering if that needs to be its own issue.

I'm wondering whether it would be worth removing the (not-working) journal style setting altogether, and making this fix properly globally applicable.

**Comment by rahaeli** [comment]
Yes, new bug for the journal-style icons page setting not working! 

I'd probably take the other route: fix the journal-style icons page and then convert site-skinned /icons to use the site-skin S2 like we did for comment pages.
